addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/CookieUtils.java

```diff
@@ -120,7 +120,8 @@ public static String getCookieName(String cookieHeaderValue) {
 
         int nameValuePairIdx = cookieHeaderValue.indexOf('=');
         if (nameValuePairIdx == NOT_FOUND
-                || cookieHeaderValue.indexOf('=') > cookieHeaderValue.indexOf(';')) {
+                || (cookieHeaderValue.indexOf(';') != NOT_FOUND
+                        && cookieHeaderValue.indexOf('=') > cookieHeaderValue.indexOf(';'))) {
```

Comment:
Extract a method to avoid finding the semicolon twice (we should also reuse `nameValuePairIdx`).

Reply:
Yes, it looks better. Done.